### PR TITLE
[BC5] Small improvements to v6-MIGRATION.md

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation

Making sure this `v6-MIGRATION.md` is as good as it can be for release 💪 

### Description

- Some reorganization
- More descriptions
- Small fixes

### CODE REVIEW

File: migrations/v6-MIGRATION.md [hunk 1]
```diff
@@ -30,20 +40,19 @@
 
 ### StoreProduct changes
 
-`StoreProduct` has been made an interface, which `GoogleStoreProduct` and `AmazonStoreProduct` implement. 
-`StoreProduct` is no longer `Parcelable`.
```
Comment (tonidero) [inline migrations/v6-MIGRATION.md:34]: Hmm I think it's worth mentioning this. Some devs implementations might rely on this.

File: migrations/v6-MIGRATION.md [hunk 2]
```diff
@@ -171,6 +180,8 @@ Replaces all of the following:
 
 ### Removed APIs
 
+This APIs were deprecated in V5 and are now being removed in V6.
```
Comment (tonidero) [inline migrations/v6-MIGRATION.md:183]:
```suggestion
These APIs were deprecated in V5 and are now being removed in V6.
```

**Review by tonidero — APPROVED** [review]
LGTM!

**Review by greenietea — APPROVED** [review]

File: migrations/v6-MIGRATION.md [hunk 3]
```diff
@@ -40,8 +40,6 @@ more thorough explanation of the new Google subscription model announced with Bi
 
 ### StoreProduct changes
 
-`StoreProduct` has been made an interface, which `GoogleStoreProduct` and `AmazonStoreProduct` implement.
```
Comment (swehner) [inline migrations/v6-MIGRATION.md]: This is still true, right?

**Review by swehner — APPROVED** [review]

File: migrations/v6-MIGRATION.md (hunk 3 above)
Comment (joshdholtz) [reply migrations/v6-MIGRATION.md]: Ha yup, did not mean to delete! Just added it back in 🙈

File: migrations/v6-MIGRATION.md [hunk 4]
```diff
@@ -30,20 +40,19 @@
...
 | sku                           | id                                                                            |
 | price                         | Moved to `price`                                                              |
 | priceAmountMicros             | Moved to `price`                                                              |
 | priceCurrencyCode             | Moved to `price`                                                              |
-| freeTrialPeriod               | Moved to PricingPhase                                                         |
-| introductoryPrice             | Moved to PricingPhase                                                         |
-| introductoryPriceAmountMicros | Moved to PricingPhase                                                         |
-| introductoryPricePeriod       | Moved to PricingPhase                                                         |
-| introductoryPriceCycles       | Moved to PricingPhase                                                         |
+| freeTrialPeriod               | Moved to `PricingPhase` in a `SubscriptionOption`                             |
```
Comment (beylmk) [inline migrations/v6-MIGRATION.md:51]: i wonder if we should be more descriptive here.. we don't really have the field `freeTrialPeriod` in `PricingPhase`.

maybe we should say something like: "replaced with related PricingPhase fields". and we also have the extension `SubscriptionOptions.getFreeTrial` that we could maybe call out here

**Pull request merged**